pkg/gadgets/run/types/factory.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"reflect"
+	"unsafe"
+
+	"golang.org/x/exp/constraints"
+)
+
+type EventFactory struct {
+	nextOffset uintptr
+	nextIndex  int
+
+	setters map[string]any
+}
+
+func NewEventFactory() *EventFactory {
+	return &EventFactory{
+		nextIndex: IndexFixed + 1,
+		setters:   make(map[string]any),
+	}
+}
+
+func (f *EventFactory) NewEvent() *Event {
+	ev := &Event{
+		Blob: make([][]byte, f.nextIndex),
+	}
+
+	ev.Blob[IndexFixed] = make([]byte, f.nextOffset)
+
+	return ev
+}
+
+type FieldType interface {
+	constraints.Integer | constraints.Float | bool
+}
+
+func FactoryAddField[T FieldType](f *EventFactory, name string) ColumnDesc {
+	offset := f.nextOffset
+
+	var zero T
+	typ := reflect.TypeOf(zero)
+
+	col := ColumnDesc{
+		Name: name,
+		Type: Type{
+			Kind: reflectTypeToKind(typ),
+		},
+		Offset:    offset,
+		BlobIndex: IndexFixed,
+	}
+
+	f.nextOffset += typ.Size()
+
+	f.setters[name] = func(ev *Event, v T) {
+		*(*T)(unsafe.Pointer(&ev.Blob[IndexFixed][offset])) = v
```

Comment:
To support different endianness between the client and the server:
```
binary.BigEndian.PutUint64(ev.Blob[IndexFixed][offset:offset+8], (uint64) v)
```

And fix `f.nextOffset += 8`.

Alternatively, use the correct method PutUint16, PutUint32, PutUint64 etc. (even if it does not play as nice with generics)

I wonder if it would be easier to define `FixedSizedMembers [][]byte` (i.e. each fixed-sized member would have its own array)